source/common/http/codec_client.h

```diff
@@ -219,9 +219,11 @@ class CodecClient : protected Logger::Loggable<Logger::Id::client>,
   struct ActiveRequest : LinkedObject<ActiveRequest>,
                          public Event::DeferredDeletable,
                          public StreamCallbacks,
-                         public ResponseDecoderWrapper {
-    ActiveRequest(CodecClient& parent, ResponseDecoder& inner)
-        : ResponseDecoderWrapper(inner), parent_(parent) {}
+                         public ResponseDecoderWrapper,
+                         public RequestEncoderWrapper {
+    ActiveRequest(CodecClient& parent, ResponseDecoder& inner, bool await_encode_complete)
+        : ResponseDecoderWrapper(inner), RequestEncoderWrapper(nullptr), parent_(parent),
+          encode_complete_(!await_encode_complete) {}
```

Comment:
wait how did we go from awaiting to deciding the encode was complete?
Ah on reflection I assume this is for simplicity since we're "not tracking" but it's going to confuse some deugging some day so let's not shortcut here :-)

Reply:
Ack, will improve readability.